src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -329,6 +335,123 @@ private static BoundCall CreateParameterlessCall(CSharpSyntaxNode syntax, BoundE
             { WasCompilerGenerated = true };
         }
 
+        // A synthesized entrypoint that forwards all calls to an async Main Method
+        internal sealed class AsyncForwardEntryPoint : SynthesizedEntryPointSymbol
+        {
+            // The user-defined asynchronous main method.
+            private readonly MethodSymbol _userMain;
+
+            private readonly MethodSymbol _getAwaiterMethod;
+            private readonly MethodSymbol _getResultMethod;
+
+            private readonly ImmutableArray<ParameterSymbol> _parameters;
+
+            // Task -> Void
+            // Task<_> -> Int32
+            private static TypeSymbol TranslateReturnType(CSharpCompilation compilation, TypeSymbol returnType)
+            {
+                return returnType.IsGenericTaskType(compilation)
+                    ? compilation.GetSpecialType(SpecialType.System_Int32)
+                    : compilation.GetSpecialType(SpecialType.System_Void);
```

Comment:
>compilation.GetSpecialType(SpecialType.System_Void) [](start = 22, length = 51)

It feels like this can fail without producing any errors, we should be using a helper from the Binder instead that makes sure the right diagnostics is reported.